docs/src/pages/documentation/cloudflare-workers.md

```diff
@@ -0,0 +1,375 @@
+    ├── shell-app/        # Your existing SPA (React, Vue, etc.)
+    └── fragment-app/     # Your web fragment
+```
+
+Create a `pnpm-workspace.yaml` at the root:
+
+```yaml
+packages:
+  - "packages/*"
+```
+
+### 2. Configure Your Fragment Worker
+
+Let's start with the fragment. The independent application you want to embed. Here's a typical `wrangler.toml` for a Remix-based fragment:
+
+```toml
+# packages/fragment-app/wrangler.toml
+name = "my-fragment"
+main = "./workers/app.ts"
+compatibility_date = "2025-10-28"
+
+# This worker doesn't need a public URL
+# It will only be accessed via Service Binding
+workers_dev = false
+```
+
+The key insight here is `workers_dev = false` Your fragment doesn't need its own public URL because it will be accessed through the gateway.
+
+### 3. Configure Your Gateway Worker
+
+The gateway is where the magic happens. It's the only worker with a public URL, and it coordinates everything:
+
+```toml
+# packages/gateway/wrangler.toml
+name = "my-gateway"
+main = "src/index.ts"
+compatibility_date = "2025-10-28"
+
+# This is your public entry point
+workers_dev = true
+
+# Serve your SPA's static files
+[assets]
+binding = "ASSETS"
+directory = "../shell-app/dist"
+
+# Connect to your fragment via Service Binding
+[[services]]
+binding = "FRAGMENT_SERVICE"
+service = "my-fragment"
+```
+
+Let's break down what's happening:
+
+- **Assets Binding**: Points to your shell application's build output. The gateway will serve these files directly.
+- **Service Binding**: Creates a direct connection to your fragment worker. The `binding` name (`FRAGMENT_SERVICE`) is how you'll reference it in code.
+
+### 4. Implement the Gateway
+
+Now let's look at the gateway's code. This is where Web Fragments really shines:
+
+```typescript
+// packages/gateway/src/index.ts
+import { FragmentGateway, getWebMiddleware } from "web-fragments/gateway";
+import type { Fetcher } from "@cloudflare/workers-types";
+
+export interface Env {
+	ASSETS: {
+		fetch: (request: Request) => Promise<Response>;
+	};
+	FRAGMENT_SERVICE: Fetcher;
+}
+
+export default {
+	async fetch(request: Request, env: Env, _ctx) {
+		// Create the gateway inside the fetch handler
+		// This ensures we have access to env bindings
+		const gateway = new FragmentGateway();
+
+		// Register your fragment
+		gateway.registerFragment({
+			fragmentId: "my-fragment",
+			// The endpoint is your Service Binding's fetch method
+			endpoint: env.FRAGMENT_SERVICE.fetch.bind(env.FRAGMENT_SERVICE),
+			routePatterns: [
+				// When users visit /dashboard in the shell app, route to this fragment
+				// Note: The fragment's internal routes don't need to match this path
+				// For example, your fragment might only have a "/" route, but the gateway
+				// mounts it at /dashboard. The fragment uses basename config to handle this.
+				"/dashboard/:_*",
+				// Handle fragment assets (the :_* means "match everything after")
+				"/__wf/my-fragment.workers.dev/:_*",
```

Comment:
this is awkward since you disabled workers.dev above. maybe just stick to `/__wf/my-fragment/:_*`?